test/jdk/java/awt/GraphicsDevice/CheckDisplayModes.java

```diff
@@ -35,18 +35,28 @@ public class CheckDisplayModes {
 
     public static void main(String[] args) {
         GraphicsEnvironment ge = GraphicsEnvironment.getLocalGraphicsEnvironment();
+        for (GraphicsDevice gd : ge.getScreenDevices()) {
+            System.out.println("Available screen device: " + gd.getIDstring());
+        }
+
         for (GraphicsDevice graphicDevice : ge.getScreenDevices()) {
```

Comment:
I remember that, at least in ogl, querying/initializing devices can invalidate the list of existing devices on the fly. It might be better to call getScreenDevices only once, this could(or maybe not) increase the chances to catch the bug."

Reply:
Ok, storing all devices in a local array to use it in both debug output and actual resolution testing.